src/pages/index.js

```diff
@@ -3,35 +3,117 @@ import {
   initialCards,
   config,
   profileEditButton,
+  profileEditAvatarButton,
   addNewCardButton,
   profileTitleInput,
   profileDescriptionInput,
   addNewCardFormEl,
   profileEditFormEl,
+  profileEditAvatarFormEl,
 } from "../utils/constants.js";
 import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import FormValidator from "../components/FormValidator.js";
 import UserInfo from "../components/UserInfo.js";
 import Card from "../components/Card.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
 
 const pageUserInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: "#profile-avatar",
+});
+
+const api = new Api({
+  baseURL: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "98cf51ba-481b-4eec-9343-23efbe8fbf82",
+    "Content-Type": "application/json",
+  },
+});
+
+let user = api.getUserInfo();
+let apiCards = api.getInitialCards();
+let cardSection;
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    (cardData) => {
+      imageViewerPopup.open(cardData);
+    },
+    (event) => {
+      deleteCardPopup.open();
+      deleteCardPopup.setSubmitAction(() => {
+        deleteCardPopup.setLoadingState(true, "Yes");
+        api.deleteCard(cardData._id).then(() => {
+          deleteCardPopup.setLoadingState(false, "Yes");
```

Comment:
Needs correction: Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
corrected